src/sphinxexpr.cpp

```diff
@@ -2570,9 +2570,9 @@ class Expr_Case_c : public ISphStringExpr
         int iLength = 0;
         *ppStr = nullptr;
 
-        //create CSphVector and store the value
+        // create CSphVector and store the value
         CSphVector< BYTE > pStrBuffer;
-        memcpy(pStrBuffer.AddN(iDocLen+1),pDoc,iDocLen+1);
+        memcpy(pStrBuffer.AddN(iDocLen),pDoc,iDocLen);
```

Comment:
1. We have dedicated 'Append' function in the vec for it. Usually not need to call low-level 'memcpy' at all
(i.e. pStrBuffer.Append(pDoc, iDocLen) usually enough and error-proof.)

2. Is 'pStrBuffer' is a pointer? No. So, it's name should start from 'd' - as dStrBuffer, 'data StrBuffer'.